rpcserver.go

```diff
@@ -2278,6 +2282,68 @@ func (r *rpcServer) ListPeers(ctx context.Context,
 	return resp, nil
 }
 
+// GetFeatures returns the features that a peer supports.
+func (r *rpcServer) GetFeatures(ctx context.Context,
+	in *lnrpc.GetFeaturesRequest) (*lnrpc.GetFeaturesResponse, error) {
+
+	if len(in.PubKey) == 0 {
+		return nil, fmt.Errorf("pubkey must be provided")
+	}
+
+	// If the query is for our node, get our set of enabled features.
+	if bytes.Equal(in.PubKey, r.selfNode[:]) {
+		var features []*lnrpc.Feature
+
+		// Get all the feature vectors enabled for our node.
+		vectors := r.server.featureMgr.List()
+
+		// Iterate through each vector and get append its features to the
+		// response, tracking the features we already have in the accumulated
+		// map to avoid duplication.
+		accumulated := make(map[uint32]bool)
+		for _, vector := range vectors {
+			rpcFeatures := invoicesrpc.CreateRPCFeatures(vector)
+
+			// Iterate through the features and add them to our list of features
+			// if they are not already there.
+			for _, feature := range rpcFeatures {
+				// Skip the feature if we already have it.
+				if _, ok := accumulated[feature.Bit]; ok {
+					continue
+				}
+
+				// Add the feature and mark it as added in the accumulated map.
+				features = append(features, feature)
+				accumulated[feature.Bit] = true
+			}
+		}
+
+		return &lnrpc.GetFeaturesResponse{
+			Features: features,
+		}, nil
+	}
+
+	var featureVector *lnwire.FeatureVector
+
+	// Try to find the peer in our list of peers.
+	for _, serverPeer := range r.server.Peers() {
+		pubKey := serverPeer.PubKey()
+		if bytes.Equal(in.PubKey, pubKey[:]) {
+			featureVector = serverPeer.remoteFeatures
+			break
+		}
+	}
+
```

Comment:
we could also try to find the node in our graph, and Merge any features we find there